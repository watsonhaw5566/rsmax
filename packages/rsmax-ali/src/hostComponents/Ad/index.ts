import { createHostComponent } from '@rsmax/runtime';
import type React from 'react';

export interface AdProps {
  /** 广告展位码 */
  unitId: string;
  /** 否	广告查询成功的回调 */
  onLoad?: (e: any) => void;
  /** 广告查询失败的回调 */
  onError?: (e: any) => void;
}

/**
 * Ad 广告组件
 * https://opendocs.alipay.com/mini/0drov4?pathHash=2b0e72a3
 */
export const Ad: React.ComponentType<AdProps> = createHostComponent<AdProps>('ad');
