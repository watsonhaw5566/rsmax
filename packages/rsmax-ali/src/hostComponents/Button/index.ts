import React from 'react';
import { createHostComponent } from '@rsmax/runtime';

export interface ButtonProps {
  readonly dataset?: DOMStringMap;
  id?: string;
  className?: string;
  style?: React.CSSProperties;
  size?: 'default' | 'mini';
  type?: 'primary' | 'default' | 'warn';
  plain?: boolean;
  disabled?: boolean;
  loading?: boolean;
  hoverClassName?: string;
  hoverStartTime?: number;
  hoverStayTime?: number;
  hoverStopPropagation?: boolean;
  formType?: 'submit' | 'reset';
  openType?: 'share' | 'getAuthorize' | 'contactShare' | 'lifestyle' | 'chooseAvatar';
  scope?: 'phoneNumber' | 'userInfo';
  appParameter?: string;
  publicId?: string;
  onTap?: (e: any) => void;
  onClick?: (e: any) => void;
  onGetAuthorize?: (e: any) => void;
  onChooseAvatar?: (e: any) => void;
  onError?: (e: any) => void;
  children?: React.ReactNode;
}

export const Button = createHostComponent<ButtonProps>('button') as React.ComponentType<ButtonProps>;
