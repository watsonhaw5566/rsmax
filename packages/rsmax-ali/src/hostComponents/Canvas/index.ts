import React from 'react';
import { createHostComponent } from '@rsmax/runtime';

export interface CanvasProps {
  readonly dataset?: DOMStringMap;
  id?: string;
  style?: React.CSSProperties;
  className?: string;
  width?: string | number;
  height?: string | number;
  disableScroll?: boolean;
  type?: string;
  onTap?: (e: any) => void;
  onTouchStart?: (e: any) => void;
  onTouchMove?: (e: any) => void;
  onTouchEnd?: (e: any) => void;
  onTouchCancel?: (e: any) => void;
  onLongTap?: (e: any) => void;
  onReady?: (e: any) => void;
}

export const Canvas = createHostComponent<CanvasProps>('canvas') as React.ComponentType<CanvasProps>;
