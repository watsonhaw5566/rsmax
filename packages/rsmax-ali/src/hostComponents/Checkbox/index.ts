import { createHostComponent } from '@rsmax/runtime';

export interface CheckboxProps {
  readonly dataset?: DOMStringMap;
  id?: string;
  name?: string;
  className?: string;
  value?: string;
  checked?: boolean;
  disabled?: boolean;
  color?: string;
  onChange?: (e: any) => void;
}

export const Checkbox = createHostComponent<CheckboxProps>('checkbox') as React.ComponentType<CheckboxProps>;
