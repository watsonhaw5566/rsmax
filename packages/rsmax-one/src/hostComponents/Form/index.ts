import type React from 'react';
import createHostComponent from '../../createHostComponent';
import type { FormProps } from './props';
import defaults from './props/default';

export type { FormProps };

const Form: React.ComponentType<FormProps> = createHostComponent<FormProps>('form', null, defaults);

export default Form;
