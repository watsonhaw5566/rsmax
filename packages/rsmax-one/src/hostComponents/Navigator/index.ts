import type React from 'react';
import createHostComponent from '../../createHostComponent';
import type { NavigatorProps } from './props';
import defaults from './props/default';

export type { NavigatorProps };

const Navigator: React.ComponentType<NavigatorProps> = createHostComponent<NavigatorProps>('navigator', null, defaults);

export default Navigator;
