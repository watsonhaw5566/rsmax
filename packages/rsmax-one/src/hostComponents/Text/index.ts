import type React from 'react';
import createHostComponent from '../../createHostComponent';
import type TextProps from './props';
import defaults from './props/default';

export type { TextProps };

const Text: React.ComponentType<TextProps> = createHostComponent<TextProps>('text', null, defaults);

export default Text;
