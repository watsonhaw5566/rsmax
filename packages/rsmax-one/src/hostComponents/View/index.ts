import type React from 'react';
import createHostComponent from '../../createHostComponent';
import type ViewProps from './props';
import defaults from './props/default';

export type { ViewProps };

const View: React.ComponentType<ViewProps> = createHostComponent<ViewProps>('view', null, defaults);

export default View;
