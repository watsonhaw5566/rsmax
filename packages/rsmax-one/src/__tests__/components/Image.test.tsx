import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import { Image } from '../../hostComponents';

describe('Image', () => {
  it('render correctly in ali', () => {
    process.env.RSMAX_PLATFORM = 'ali';
    const testRenderer = TestRenderer.create(
      <Image
        className="class"
        onError={() => {
          // ignore
        }}
        onLoad={() => {
          // ignore
        }}
      />
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('render correctly in wechat', () => {
    process.env.RSMAX_PLATFORM = 'wechat';
    const testRenderer = TestRenderer.create(
      <Image
        className="class"
        onError={() => {
          // ignore
        }}
        onLoad={() => {
          // ignore
        }}
      />
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('render correctly in toutiao', () => {
    process.env.RSMAX_PLATFORM = 'toutiao';
    const testRenderer = TestRenderer.create(
      <Image
        className="class"
        onError={() => {
          // ignore
        }}
        onLoad={() => {
          // ignore
        }}
      />
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
