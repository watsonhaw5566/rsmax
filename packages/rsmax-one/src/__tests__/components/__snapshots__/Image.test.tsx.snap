// Rstest Snapshot v1

exports[`Image > render correctly in ali 1`] = `
<image
  className="class"
  onError={[Function]}
  onLoad={[Function]}
/>
`;

exports[`Image > render correctly in toutiao 1`] = `
<image
  className="class"
  onError={[Function]}
  onLoad={[Function]}
/>
`;

exports[`Image > render correctly in wechat 1`] = `
<image
  className="class"
  onError={[Function]}
  onLoad={[Function]}
/>
`;

exports[`Image render correctly in ali 1`] = `
<image
  className="class"
  onError={[Function]}
  onLoad={[Function]}
/>
`;

exports[`Image render correctly in toutiao 1`] = `
<image
  className="class"
  onError={[Function]}
  onLoad={[Function]}
/>
`;

exports[`Image render correctly in wechat 1`] = `
<image
  className="class"
  onError={[Function]}
  onLoad={[Function]}
/>
`;
